sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java

```diff
@@ -24,7 +24,7 @@
 /**
  * This is core Transport/Connection agnostic request to the Azure Cosmos DB database service.
  */
-public class RxDocumentServiceRequest {
+public class RxDocumentServiceRequest implements Cloneable {
```

Comment:
I know we have a `clone()` method in this class, but usually its not recommended to use `clone()` and `Cloneable` interface, as it is flawed. 
https://stackoverflow.com/a/11654256
I doubt if we are actually calling this clone method, worth checking and removing it if not being called.

Reply:
it seems we're using it here https://github.com/Azure/azure-sdk-for-java/blob/12acd7a1e6779f39fde8fdfccc126437637d61e6/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java#L126